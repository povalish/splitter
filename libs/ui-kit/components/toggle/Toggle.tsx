import React, { useId } from 'react';
import { IWebToggle } from './Toggle.interface';
import {
  container,
  title,
  input,
  toggleArea,
  toggleThumb,
  checkedTip,
  uncheckedTip,
} from './Toggle.classes';

//
//

export const Toggle: React.FC<IWebToggle> = ({ label, className, ...inputProps }) => {
  const id = useId();

  return (
    <label
      htmlFor={id}
      className={container({ className })}
      tabIndex={0}
      data-testid='toggle-input'
    >
      <span className={title()}>{label}</span>

      <input id={id} type='checkbox' className={input()} {...inputProps} />

      <div className={toggleArea()} />
      <span className={checkedTip()}>On</span>
      <div className={toggleThumb()} />
      <span className={uncheckedTip()}>Off</span>
    </label>
  );
};
