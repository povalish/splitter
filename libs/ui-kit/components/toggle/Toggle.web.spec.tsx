import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Toggle } from './Toggle';

//
//

describe('Toggle component', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Toggle label='Toggle' />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call onChange', () => {
    const onChange = jest.fn();
    render(<Toggle label='Toggle' checked={false} onChange={(e) => onChange(e.target.checked)} />);

    fireEvent.click(screen.getByTestId('toggle-input'));
    expect(onChange).toBeCalledWith(true);
  });
});
