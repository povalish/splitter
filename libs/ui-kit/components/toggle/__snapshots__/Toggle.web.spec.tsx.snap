// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle component should render successfully 1`] = `
<body>
  <div>
    <label
      class="
  relative
  hover:cursor-pointer
  group
  focus:outline-none
"
      data-testid="toggle-input"
      for=":r0:"
      tabindex="0"
    >
      <span
        class="text-colorText text-sm mb-2 group-focus-within:text-colorAccent"
      >
        Toggle
      </span>
      <input
        class="sr-only peer"
        id=":r0:"
        type="checkbox"
      />
      <div
        class="
  w-[85px] h-[40px] rounded-[54px] py-2 px-3
  bg-colorBackgroundContrast
  flex items-center justify-between
  group-focus-within:border-[3px] group-focus-within:border-colorAccent 
  group-focus-within:shadow-focus
"
      />
      <span
        class="
  text-sm font-bold
  transition-all
  absolute bottom-2.5 left-3 
  text-colorSuccess
  opacity-0 peer-checked:opacity-100
  select-none
"
      >
        On
      </span>
      <div
        class="
  w-[28px] h-[28px] rounded-full
  absolute
  bottom-[6px] left-[8px] bg-colorDisabled
  peer-checked:translate-x-10 peer-checked:bg-colorSuccess
  transition-all
"
      />
      <span
        class="
  text-sm font-bold
  transition-all
  absolute bottom-2.5 right-3 
  text-colorDisabled
  opacity-100 peer-checked:opacity-0
  select-none
"
      >
        Off
      </span>
    </label>
  </div>
</body>
`;
